### ISSUE TITLE
Spell LoS issues when inside walls (PvE)(Warrior Demo/TClap)

### ISSUE THREAD

**Issue by Godmatik** [Atlantiss/NetherwingBugtracker#8205]
**Description**: 
Some spells that fail to land when clipped into a wall

**Current behaviour**:
If someone is partially clipped into a wall and attempts to use an AoE spell like Demo shout and Thunder clap, it will not be applied and won't even show it hit the mob/boss. We suspect it has something to do with the tank having some weird LoS issues that is preventing it. 

Good example we noticed this are on Prince and Nightbane fights in Karazhan, Our tank was up against a wall tanking them, but if the tank was a bit too much backed into the wall, their Demo shout and Thunder clap abilities would not get applied when used. He was Tauren if that has any relevance.

**Expected behaviour**: 
As long as you are LoS of the boss/mob to attack and be attacked, then spells should work as well

**Server Revision**: 
3709

**Comment by Wolffenstein** [comment]
Can you provide a video/screenshot/something i can test?

**Comment by Godmatik** [comment]
Our tank will DM one to you

**Comment by MecAtlantiss** [comment]
I'm able to reproduce this bug consistently on Nightbane and Prince. 

Nightbane: https://www.twitch.tv/videos/1767890236

Prince: https://www.twitch.tv/videos/1767907159

My guess is that AoE abilities like Thunder Clap and Demoralizing Shout calculate distance to enemy based on center of player's hixbox to closest edge of enemy's hitbox. If the closest edge of the enemy's hitbox is clipping through a wall, such as what is clearly happening in the Nightbane video I linked, then perhaps the Thunder Clap doesn't view the target a valid target due to being "behind a wall". In the case of Prince, even though the visual red circle for his hitbox doesn't appear to be clipping through the wall, I suspect his true hitbox is larger than the red circle players see.

I tried to replicate the bug on Magtheridon, but couldn't.

This bug was also happening on Karazhan server.
